### DISCUSSION TITLE
Wasm build stopped working - dist directory not created

### DISCUSSION THREAD

**Discussion by borislutskovsky** [unoplatform/uno#9843] [General]
After a few weeks of active development and things working correctly, my Wasm build stopped working and creating the bin/Debug/net6.0/dist directory with the Wasm application. The last thing I remember doing was creating a local publish profile for the Wasm app. 

The application works for another developer, so I'm sure it's not a code issue, but a configuration issue. Even a new project using the Uno template has the exact same issue. 

I've tried uninstalling and reinstalling Visual Studio, the workflows, I've updated uno-check and rerun it, so far the behavior hasn't changed. I've tried switching to use Visual Studio Code and following the VS Code example. Still the same issue.

**Comment by jeromelaban** [comment]
Thanks for reaching out! Can you [provide a binlog](https://platform.uno/docs/articles/uno-builds-troubleshooting.html) of your app's build?

**Reply by borislutskovsky** [reply]
[binlog.zip](https://github.com/unoplatform/uno/files/9568601/binlog.zip)
here you go

**Reply by jeromelaban** [reply]
Thanks. For some reason, it looks like your nuget cache is corrupted and [this file](https://github.com/unoplatform/Uno.Wasm.Bootstrap/blob/main/src/Uno.Wasm.Bootstrap/build/Uno.Wasm.Bootstrap.targets) does not contain what it's supposed to have.

I'd suggest to [clear all nuget caches ](https://docs.microsoft.com/en-us/nuget/consume-packages/managing-the-global-packages-and-cache-folders#clearing-local-folders) and do a rebuild: 
```
dotnet nuget locals all --clear
```

**Reply by borislutskovsky** [reply]
that was semi-productive. 

I'm now seeing the following build errors. I don't understand why it's trying to use 5.x if the project is targeting 6.x.

C:\Users\BorisLutskovsky\.nuget\packages\uno.wasm.bootstrap\3.3.1\build\Uno.Wasm.Bootstrap.targets(175,5): error : [00:
00:00.0274339] Cannot use file stream for [C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.17\Microsoft.NETCor
e.App.deps.json]: No such file or directory [C:\Users\BorisLutskovsky\source\repos\App1\App1\App1.Wasm\App1.Wasm.csproj
]
C:\Users\BorisLutskovsky\.nuget\packages\uno.wasm.bootstrap\3.3.1\build\Uno.Wasm.Bootstrap.targets(175,5): error : [00:
00:00.0281875] A fatal error was encountered. The library 'hostpolicy.dll' required to execute the application was not
found in 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\5.0.17'. [C:\Users\BorisLutskovsky\source\repos\App1\App
1\App1.Wasm\App1.Wasm.csproj]
C:\Users\BorisLutskovsky\.nuget\packages\uno.wasm.bootstrap\3.3.1\build\Uno.Wasm.Bootstrap.targets(175,5): error : Syst
em.Exception: Failed to generate wasm layout (More details are available in diagnostics mode or using the MSBuild /bl s
witch) [C:\Users\BorisLutskovsky\source\repos\App1\App1\App1.Wasm\App1.Wasm.csproj]
C:\Users\BorisLutskovsky\.nuget\packages\uno.wasm.bootstrap\3.3.1\build\Uno.Wasm.Bootstrap.targets(175,5): error :    a
t Uno.Wasm.Bootstrap.ShellTask_vfc89b40db92a14b8f04d4c5c9c72690bdcd7b878.RunPackager() in D:\a\1\s\src\Uno.Wasm.Bootstr
ap\ShellTask.cs:line 746 [C:\Users\BorisLutskovsky\source\repos\App1\App1\App1.Wasm\App1.Wasm.csproj]
C:\Users\BorisLutskovsky\.nuget\packages\uno.wasm.bootstrap\3.3.1\build\Uno.Wasm.Bootstrap.targets(175,5): error :    a
t Uno.Wasm.Bootstrap.ShellTask_vfc89b40db92a14b8f04d4c5c9c72690bdcd7b878.Execute() in D:\a\1\s\src\Uno.Wasm.Bootstrap\S
hellTask.cs:line 228 [C:\Users\BorisLutskovsky\source\repos\App1\App1\App1.Wasm\App1.Wasm.csproj]
    1 Warning(s)
    3 Error(s)

**Reply by borislutskovsky** [reply]
[binlog2.zip](https://github.com/unoplatform/uno/files/9568750/binlog2.zip)
here's the binlog for the latest attempt to build.

**Reply by jeromelaban** [reply]
Ok, this issue is a bit different, and it's related to an implicit dependency on .NET 5. Can you try installing the [.NET 5 SDK](https://dotnet.microsoft.com/en-us/download/dotnet/5.0) and see if it helps?  We'll track where this dependency come from.

**Reply by borislutskovsky** [reply]
Thank you. that appears to have succeeded and created a dist directory.
